python/tests/test_schema.py

```diff
@@ -671,3 +672,24 @@ def test_prune_columns_struct_in_map_full():
 def test_prune_columns_select_original_schema(table_schema_nested: Schema):
     ids = set(range(table_schema_nested.highest_field_id))
     assert prune_columns(table_schema_nested, ids, True) == table_schema_nested
+
+
+def test_schema_select(table_schema_nested: Schema):
+    assert table_schema_nested.select("bar", "baz") == Schema(
+        NestedField(field_id=2, name="bar", field_type=IntegerType(), required=True),
+        NestedField(field_id=3, name="baz", field_type=BooleanType(), required=False),
+        schema_id=1,
+        identifier_field_ids=[1],
```

Comment:
Minor: Should this be updated? This schema doesn't have all of the identifier field IDs, so maybe we should clear the identifier field ID list?

Reply:
Great catch, I'll now take the intersect of the `identifier_field_ids` and the IDs of the fields